buildSrc/src/main/groovy/com/jakewharton/rxbindingproject/ValidateBindingsTask.groovy

```diff
@@ -0,0 +1,87 @@
+package com.jakewharton.rxbindingproject
+
+import com.github.javaparser.JavaParser
+import com.github.javaparser.ast.CompilationUnit
+import com.github.javaparser.ast.body.MethodDeclaration
+import com.github.javaparser.ast.body.Parameter
+import com.github.javaparser.ast.type.ReferenceType
+import com.github.javaparser.ast.type.WildcardType
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter
+import com.google.common.collect.ImmutableList
+import org.gradle.api.tasks.SourceTask
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.incremental.IncrementalTaskInputs
+
+class ValidateBindingsTask extends SourceTask {
+
+  private static final List<String> METHOD_ANNOTATIONS = ImmutableList.of(
+      "CheckResult",
+      "NonNull"
+  )
+
+  @TaskAction
+  def validate(IncrementalTaskInputs inputs) {
+    getSource().each { verifyJavaFile(it) }
+  }
+
+  static void verifyJavaFile(File javaFile) {
+    CompilationUnit cu = JavaParser.parse(javaFile)
+
+    cu.accept(new VoidVisitorAdapter<Object>() {
+      @Override
+      void visit(MethodDeclaration n, Object arg) {
+        verifyMethodAnnotations(n)
+        verifyParameters(n)
+        verifyReturnType(n)
+        // Explicitly avoid going deeper, we only care about top level methods. Otherwise
+        // we'd hit anonymous inner classes and whatnot
```

Comment:
You don't need to change all the comments in here, but in the future be mindful that we like [comments to be sentences][1]. (You're missing a period here and a lot of other places)

 [1]: http://nedbatchelder.com/blog/201401/comments_should_be_sentences.html